Source/SwiftLintFramework/Rules/Lint/BalancedXCTestLifecycleRule.swift

```diff
@@ -113,16 +113,31 @@ public struct BalancedXCTestLifecycleRule: SwiftSyntaxRule, OptInRule, Configura
     // MARK: - Public
 
     public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
-        Visitor(viewMode: .sourceAccurate)
+        Visitor(viewMode: .sourceAccurate, testClasses: configuration.testParentClasses)
+    }
+
+    public func makeViolation(file: SwiftLintFile, violation: ReasonedRuleViolation) -> StyleViolation {
```

Comment:
Can be removed if you make `UnitTestRuleConfiguration` conform to `SeverityBasedRuleConfiguration` instead of `RuleConfiguration` only.

Reply:
Thanks for the pointers - having to reimplement `makeViolation` was starting to bother me. I will try and action these later on today ...